Seymour/Matrix/TotalUnimodularityTest.lean

```diff
@@ -48,6 +48,14 @@ theorem Matrix.testTotallyUnimodular_eq_isTotallyUnimodular {m n : ℕ} (A : Mat
 instance {m n : ℕ} (A : Matrix (Fin m) (Fin n) ℚ) : Decidable A.IsTotallyUnimodular :=
   decidable_of_iff _ A.testTotallyUnimodular_eq_isTotallyUnimodular
 
+def Matrix.square_set_submatrix {m n : ℕ} (A : Matrix (Fin m) (Fin n) ℚ)
```

Comment:
If I remember correctly, @Ivan-Sergeyev was trying something similar earlier and it was too cumbersome to work with. However, I don't have any strong opinion on whether it should be used here.